lib/Interpreter/DeclUnloader.cpp

```diff
@@ -711,10 +736,49 @@ bool DeclUnloader::VisitRedeclarable(clang::Redeclarable<T>* R, DeclContext* DC)
 
   bool DeclUnloader::VisitNamespaceDecl(NamespaceDecl* NSD) {
     // NamespaceDecl: NamedDecl, DeclContext, Redeclarable
-    bool Successful = VisitDeclContext(NSD);
-    Successful &= VisitRedeclarable(NSD, NSD->getDeclContext());
+
+    // When unloading a NamedDecl within an inline namespace within std the
+    // NamedDecl needs to be removed from the parent std as well.
+    // Not sure why, but test/CodeUnloading/NameSpaces.C demonstrates
```

Comment:
@marsupial: the error shows that the lookup succeeded but the linker cannot resolve the address because it is gone.

Here is how the error should look like after unloading:
```
[cling]$ .rawInput
Using raw input
[cling]! namespace A{}
[cling]! auto var1 = A::f;
input_line_4:1:16: error: no member named 'f' in namespace 'A'
auto var1 = A::f;
            ~~~^
[cling]!
```